widgets-raster/index.ts

```diff
@@ -0,0 +1,300 @@
+        values: [value],
+        owner: TREE_WIDGET_ID
+      });
+      render({includeTreemap: false}); // this makes sure the treemap zoom function works by not rendering the treemap again
+    } else {
+      removeFilter(filters, {column: CATEGORY_COLUMN, owner: TREE_WIDGET_ID});
+      render();
+    }
+  }
+});
+
+const clearBtn = document.querySelector('.clear-btn') as HTMLButtonElement;
+clearBtn.addEventListener('click', () => {
+  removeFilter(filters, {column: CATEGORY_COLUMN});
+  clearBtn.style.visibility = 'hidden';
+  render();
+});
+
+const getFillColorLayer = (d: GeoJSON.Feature<GeoJSON.Geometry, {band_1: number}>) => {
+  const value = d.properties.band_1;
+  if (rasterMetadata) {
+    const meta = rasterMetadata.bands[0];
+    if (meta.colorinterp === 'palette') {
+      const category = meta.colortable?.[value];
+      if (category) {
+        const [r, g, b] = category;
+        if (r === 0 && g === 0 && b === 0) {
+          return [0, 0, 0, 0] as Color;
+        }
+
+        return category as Color;
+      }
+    }
+  }
+  return [0, 0, 0, 0] as Color;
+};
+
+function renderLayers() {
+  const layers = [
+    new RasterTileLayer({
+      id: 'raster_layer',
+      data: source,
+      pickable: true,
+      getFillColor: getFillColorLayer,
+      onDataLoad: (tilejson: any) => {
+        rasterMetadata = tilejson.raster_metadata;
+      },
+      onViewportLoad: async tiles => {
+        const {widgetSource} = await source;
+        await widgetSource.loadTiles(tiles);
+        debouncedRenderWidgets();
+      },
+      extensions: [new DataFilterExtension({filterSize: 4})],
+      ...getDataFilterExtensionProps(filters)
+    })
+  ];
+
+  deck.setProps({
+    layers,
+    getTooltip: ({object}) => {
+      const value = object?.properties?.band_1;
+      if (!value) {
+        return null;
+      }
+
+      return {
+        text: RASTER_CATEGORY_MAP[value]
+      };
+    }
+  });
+}
+
+// WIDGETS
+
+// Loading state and handlers for widgets
+
+let widgetsLoading = false;
+
+function setWidgetsLoading() {
+  if (widgetsLoading) {
+    return;
+  }
+
+  widgetsLoading = true;
+  formulaWidget.innerHTML = '<span style="font-weight: 400; font-size: 14px;">Loading...</span>';
+  treemapChart.showLoading();
+}
+
+// Render widgets
+
+async function renderWidgets({includeTreemap = true} = {}) {
+  if (!source) return;
+  const {widgetSource} = await source;
+
+  if (includeTreemap) {
+    await renderTreemap(widgetSource);
+  } else {
+    treemapChart.hideLoading();
+  }
+
+  formulaWidget.innerHTML = '<span style="font-weight: 400; font-size: 14px;">Loading...</span>';
+  await renderFormula(widgetSource);
+  widgetsLoading = false;
+}
+
+async function renderFormula(ws: WidgetSource<WidgetSourceProps>) {
+  const formula = await ws.getFormula({
+    column: '*',
+    operation: 'count',
+    filters: filters,
+    spatialFilter: getSpatialFilterFromViewState(viewState),
+    spatialIndexReferenceViewState: viewState
```

Comment:
The `spatialIndexReferenceViewState` parameter is no longer required or available in v0.5.0, this can be removed here and below!

Reply:
Done!